app/views/responses/show.html.erb

```diff
@@ -21,9 +21,35 @@
         </div>
         <p style="font-size:0.7rem">*Note: You will need to sign in/up before you can edit your response*</p>
         <% if @chomp_session.user == current_user %>
-          <%= link_to "Generate Results", chomp_session_result_path, class: "btn btn-info btn-lg", id: "bootstrap-overrides" %>
+          <button type="button" class="btn btn-info btn-lg" data-toggle="modal" data-target="#exampleModalCenter">
+            Generate Results
+          </button>
         <% end %>
       </div>
     </div>
   </div>
 </div>
+
+<!-- Button trigger modal -->
+
+
+<!-- Modal -->
+<div class="modal fade" id="exampleModalCenter" tabindex="-1" role="dialog" aria-labelledby="exampleModalCenterTitle" aria-hidden="true">
+  <div class="modal-dialog modal-dialog-centered" role="document">
+    <div class="modal-content">
+      <div class="modal-header">
+        <h5 class="modal-title" id="exampleModalCenterTitle">Generate Results</h5>
+        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+          <span aria-hidden="true">&times;</span>
+        </button>
+      </div>
+      <div class="modal-body">
+        Are you sure that you want to end the poll prematurely and generate your restaurant result?
+      </div>
+      <div class="modal-footer">
+        <button type="button" class="btn btn-secondary" data-dismiss="modal">Close</button>
```

Comment:
can you use btn-success or something so it's not grey? i might change the btn-secondary to orange instead if you prefer that

Reply:
Updated to btn-success